### DISCUSSION TITLE
looping tasks

### DISCUSSION THREAD

**Discussion by chriscamacho** [crewAIInc/crewAI#1147] [Q&A]
Ideally I'd like to do something like this...

1 programming agent creates code
2 testing agent runs code in repl and provides feedback given some example require output
  testing passed goes to set 4
3 programming agent takes code from step 2 re-codes and goes to step 2
4 resulting code is presented

so this is 3 tasks really, but in particular step 2 needs a way to complete the whole process and step 3 needs a way to be able to get back to step 2

**Comment by Biancamazzi** [comment]
don't forget we have https://chat.openai.com/g/g-qqTuUWsBY-crewai-assistant to help and also the discord community https://discord.com/invite/X4JWnZnxPb

**Reply by mouramax** [reply]
It seems like you have a case that fits well within the Flows paradigm. I strongly suggest you take a look at the following references:

* The CrewAI [Flows documentation here](https://docs.crewai.com/concepts/flows).
* [This video](https://www.youtube.com/watch?v=8PtGcNE01yo) and [this video](https://www.youtube.com/watch?v=HXP3u_D_xSY), by @bhancockio. They're incredible videos on the subject, including coverage of feedback loops.

**Comment by Kolaposki** [comment]
Check out the ``task.execute()`` method.

That's what I use to get the output of a task and then move on to the next task or perform some operations. You can also use task.output which simply just gives you the result of the executed task.

**Reply by ZvikaZ** [reply]
Can you refer to where `task.execute()` is mentioned in the documentation?

**Reply by BokisAngelov** [reply]
Check this repo here - https://github.com/apappascs/crewai-independent-execution

Have in mind that if you use task.execute(), you are executing only the specific task without continuing your crew execution.

**Comment by Toppbeatz** [comment]
I been trying to figure out how to do the same thing.
